src/jupyter_contrib_nbextensions/nbextensions/pyVariableInspector/main.js

```diff
@@ -0,0 +1,397 @@
+    'base/js/utils', 'notebook/js/codecell'
+], function(require, $, IPython, configmod, utils, codecell) {
+
+    var Notebook = require('notebook/js/notebook').Notebook 
+    "use strict";
+    //require('https://cdnjs.cloudflare.com/ajax/libs/jquery.tablesorter/2.25.7/js/jquery.tablesorter.min.js')
+
+    // ...........Parameters configuration......................
+    // define default values for config parameters if they were not present in general settings (notebook.json)
+    var cfg = {
+        'window_display': false,
+        'lenName': 16,
+        'lenType': 16,
+        'lenVar': 40,
+        'types_to_exclude': ['module', 'function', 'builtin_function_or_method', 'instance', '_Feature']  
+    }
+
+    //.....................global variables....
+
+    var liveNotebook = !(typeof IPython == "undefined")
+
+    var st = {}
+    st.config_loaded = false;
+    st.extension_initialized = false;
+
+    function read_config(cfg, callback) { // read after nb is loaded
+        // create config object to load parameters
+        var base_url = utils.get_body_data("baseUrl");
+        var initial_cfg = $.extend(true, {}, cfg);
+        var config = new configmod.ConfigSection('notebook', { base_url: base_url });
+        config.loaded.then(function() {
+            // config may be specified at system level or at document level.
+            // first, update defaults with config loaded from server
+            cfg = $.extend(true, cfg, config.data.pyVarInspector);
+            // then update cfg with anything found in current notebook metadata
+            // and save in nb metadata (then can be modified per document)
+            if (IPython.notebook.metadata.pyVarInspector) {
+                if (IPython.notebook.metadata.pyVarInspector.window_display)
+                    cfg.window_display = IPython.notebook.metadata.pyVarInspector.window_display;
+            }
+            cfg = IPython.notebook.metadata.pyVarInspector = $.extend(true, cfg,
+            IPython.notebook.metadata.pyVarInspector);
+            
+
+            if (config.data.pyVarInspector.types_to_exclude){
+                cfg.types_to_exclude = config.data.pyVarInspector.types_to_exclude
+            }
+            // call callbacks
+            callback && callback();
+            st.config_loaded = true;
+        })
+        config.load();
+        return cfg;
+    }
+
+    var sortable;
+    var bst = require(
+        ['https://cdnjs.cloudflare.com/ajax/libs/jquery.tablesorter/2.25.7/js/jquery.tablesorter.min.js'],
+        function() {
+            sortable = true;
+        },
+        function(err) {
+            sortable = false;
+        }
+    );
+
+    function toggleVarInspector() {
+        toggle_varInspector(cfg, st)
+    }
+
+    var varInspector_button = function() {
+        if (!IPython.toolbar) {
+            $([IPython.events]).on("app_initialized.NotebookApp", varInspector_button);
```

Comment:
accessing events from the global namespace is (I think) deprecated - you should add `base/js/events` to the `define([` requirement list, and then call `events.on`